src/v2/view-builder/views/EnrollProfileView.js

```diff
@@ -61,8 +61,28 @@ export default BaseView.extend({
   Body,
   Footer,
   createModelClass(currentViewState, optionUiSchemaConfig, settings) {
-    let ModelClass = BaseView.prototype.createModelClass.apply(this, arguments);
     const currentSchema = JSON.parse(JSON.stringify((currentViewState.uiSchema)));
+    let ModelClass = BaseView.prototype.createModelClass.apply(this, arguments, currentViewState);
+
+    ModelClass = ModelClass.extend({
+      toJSON: function() {
+        const modelJSON = Model.prototype.toJSON.call(this, arguments, currentViewState);
+        // remove unwanted data from modelJSON
```

Comment:
change the comment to `delete optional attributes if they are empty and not required`